src/v2/Apps/Artist/Routes/AuctionResults/__tests__/ArtistAuctionResults.jest.tsx

```diff
@@ -75,84 +55,121 @@ describe("AuctionResults", () => {
+  })
+
+  const { renderWithRelay } = setupTestWrapperTL({
+    Component: (props: any) => (
+      <MockBoot breakpoint={breakpoint}>
+        <AuctionResultsRoute {...props} />
+      </MockBoot>
+    ),
+    query: graphql`
+      query ArtistAuctionResults_Test_Query($artistID: String!)
+        @raw_response_type {
+        artist(id: $artistID) {
+          ...ArtistAuctionResultsRoute_artist
+        }
+      }
+    `,
+    variables: {
+      artistID: "pablo-picasso",
+    },
+  })
+
   describe("trigger auth modal for filtering and pagination", () => {
-    beforeEach(async () => {
-      wrapper = await getWrapper()
-    })
     afterEach(() => {
       mockOpenAuthModal.mockReset()
     })
 
-    it("calls auth modal for 1st pagination but not for 2nd", done => {
-      const pagination = wrapper.find(Pagination)
-      pagination.find("a").at(1).simulate("click")
+    it("calls auth modal for 1st pagination but not for 2nd", () => {
+      renderWithRelay(mockedResolver)
+      const navigation = screen.getByRole("navigation")
+      const links = within(navigation).getAllByRole("link")
+      expect(links).toHaveLength(6)
 
-      setTimeout(() => {
-        expect(mockOpenAuthModal).toHaveBeenCalledTimes(1)
-      })
+      fireEvent.click(links[2])
+      expect(mockOpenAuthModal).toHaveBeenCalledTimes(1)
+      fireEvent.click(links[3])
+      expect(mockOpenAuthModal).toHaveBeenCalledTimes(1)
+    })
 
-      pagination.find("a").at(2).simulate("click")
+    it("calls auth modal for 1st category selection but not for 2nd", () => {
+      let operationVariables
+      const { env } = renderWithRelay(mockedResolver, true)
 
-      setTimeout(() => {
-        // expect no new call
-        expect(mockOpenAuthModal).toHaveBeenCalledTimes(1)
-        done()
+      act(() => {
+        env.mock.resolveMostRecentOperation(operation => {
+          return MockPayloadGenerator.generate(operation, mockedResolver)
+        })
       })
-    })
 
-    it("calls auth modal for 1st medium selection but not for 2nd", done => {
-      const filter = wrapper.find("MediumFilter")
-      const checkboxes = filter.find("Checkbox")
-
-      checkboxes.at(1).simulate("click")
-      setTimeout(() => {
-        expect(openAuthModal).toHaveBeenCalledTimes(1)
+      const checkboxes = screen.getAllByRole("checkbox")
+      fireEvent.click(checkboxes[1])
+      act(() => {
+        env.mock.resolveMostRecentOperation(operation => {
+          operationVariables = operation.request.variables
+          return MockPayloadGenerator.generate(operation, mockedResolver)
+        })
       })
 
-      checkboxes.at(2).simulate("click")
-      setTimeout(() => {
-        // expect no new call
-        expect(openAuthModal).toHaveBeenCalledTimes(1)
-        done()
+      expect(operationVariables.categories).toContain("Work on Paper")
+      expect(openAuthModal).toHaveBeenCalledTimes(1)
+
+      fireEvent.click(checkboxes[2])
+      act(() => {
```

Comment:
instead of spying relay.refetch prop this is the way we should observe for new relay operations, if we don't receive a second one will throw an error. With this approach, we're able to check changes from the relay query, in this case if the query was updated with the selected filter.